08_banco_de_dados/test/test_exercicio.rb

```diff
@@ -49,4 +48,230 @@ class TestExercicio < Minitest::Test
+        'value' => 0.5
+      },
+      {
+        'sacoles_id' => 23,
+        'flavor' => 'pitanga',
+        'value' => 1.0
+      }
+    ]
+    assert_equal sacoleResult, sacole_insert_expected
+
+    #littleshop
+    db.execute 'INSERT INTO littleshop VALUES(?, ?)', 1, 'Yummy Littleshop'
+    littleshopResult = db.execute 'SELECT * FROM littleshop'
+    littleshop_insert_expected = [
+      {
+        'littleshop_id' => 01,
+        'name' => 'Yummy Littleshop'
+      }
+    ]
+    assert_equal littleshopResult, littleshop_insert_expected
+
+    #storage
+    db.execute 'INSERT INTO storage(storage_id, littleshop_id, sacoles_id, sold) VALUES (?, ?, ?, ?)', 11, 1, 21, 0
+    db.execute 'INSERT INTO storage(storage_id, littleshop_id, sacoles_id, sold) VALUES (?, ?, ?, ?)', 12, 1, 21, 0
+    db.execute 'INSERT INTO storage(storage_id, littleshop_id, sacoles_id, sold) VALUES (?, ?, ?, ?)', 13, 1, 22, 0
+    db.execute 'INSERT INTO storage(storage_id, littleshop_id, sacoles_id, sold) VALUES (?, ?, ?, ?)', 14, 1, 22, 0
+    db.execute 'INSERT INTO storage(storage_id, littleshop_id, sacoles_id, sold) VALUES (?, ?, ?, ?)', 15, 1, 23, 0
+
+    storageResult = db.execute 'SELECT * FROM storage'
+    storage_insert_expected = [
+      {
+        'storage_id' => 11,
+        'littleshop_id' => 1,
+        'sacoles_id' => 21,
+        'sold' => 0, 
+      },
+      {
+        'storage_id' => 12,
+        'littleshop_id' => 1,
+        'sacoles_id' => 21,
+        'sold' => 0, 
+      },
+      {
+        'storage_id' => 13,
+        'littleshop_id' => 1,
+        'sacoles_id' => 22,
+        'sold' => 0, 
+      },
+      {
+        'storage_id' => 14,
+        'littleshop_id' => 1,
+        'sacoles_id' => 22,
+        'sold' => 0, 
+      },
+      {
+        'storage_id' => 15,
+        'littleshop_id' => 1,
+        'sacoles_id' => 23,
+        'sold' => 0, 
+      },
+    ]
+
+    assert_equal storageResult.size, 5
+    assert_equal storageResult, storage_insert_expected
+  end
+
+
+  def test_sell_sacole
+    db = SQLite3::Database.open 'sacole.db'
+    db.results_as_hash = true
+    #TABLES CREATION
+
+    # test
+    db.execute 'DROP TABLE IF EXISTS littleshop'
+    db.execute 'DROP TABLE IF EXISTS sacoles'
+    db.execute 'DROP TABLE IF EXISTS storage'
+
+    # littleshop
+    db.execute 'CREATE TABLE littleshop(littleshop_id INTEGER PRIMARY KEY AUTOINCREMENT, name TEXT)'
+    
+    # sacole
+    db.execute 'CREATE TABLE sacoles(sacoles_id INTEGER PRIMARY KEY AUTOINCREMENT, flavor TEXT, value REAL)'
+
+    # storage
+    db.execute 'PRAGMA foreign_keys = ON'
+    resultkey = db.execute 'PRAGMA foreign_keys'
+    assert_equal [{'foreign_keys' => 1}], resultkey
+    db.execute 'CREATE TABLE storage(storage_id INTEGER PRIMARY KEY AUTOINCREMENT, littleshop_id INTEGER, sacoles_id INTEGER, sold BOOLEAN, FOREIGN KEY (littleshop_id) REFERENCES littleshop(littleshop_id), FOREIGN KEY (sacoles_id) REFERENCES sacoles(sacoles_id))'
```

Comment:
[Not blocked]

Como esse código se repete em todos os testes, uma boa ideia é extrair ele para um método para que cada teste chame ele novamente.

Exemplo:
```ruby
def setup_db
db = SQLite3::Database.open 'sacole.db'
    db.results_as_hash = true
    #TABLES CREATION

    # test
    db.execute 'DROP TABLE IF EXISTS littleshop'
    db.execute 'DROP TABLE IF EXISTS sacoles'
    db.execute 'DROP TABLE IF EXISTS storage'

    # littleshop
    db.execute 'CREATE TABLE littleshop(littleshop_id INTEGER PRIMARY KEY AUTOINCREMENT, name TEXT)'

    # sacole
    db.execute 'CREATE TABLE sacoles(sacoles_id INTEGER PRIMARY KEY AUTOINCREMENT, flavor TEXT, value REAL)'

    # storage
    db.execute 'PRAGMA foreign_keys = ON'
    resultkey = db.execute 'PRAGMA foreign_keys'
    assert_equal [{'foreign_keys' => 1}], resultkey
    db.execute 'CREATE TABLE storage(storage_id INTEGER PRIMARY KEY AUTOINCREMENT, littleshop_id INTEGER, sacoles_id INTEGER, sold BOOLEAN, FOREIGN KEY (littleshop_id) REFERENCES littleshop(littleshop_id), FOREIGN KEY (sacoles_id) REFERENCES sacoles(sacoles_id))'

db
end
```

E no teste:

```suggestion
    db = setup_db
```